### PR TITLE
feat(agw): Add interface for tracing, implement interface against sen…

### PR DESCRIPTION
**Pull request by mstre123**
…try-go.

Signed-off-by: mstre123 <mstre123@gmail.com>

## Summary

Interface for tracing in AGW

## Test Plan

test locally

## Additional Information

Will fill out additional tests, comments etc. Wanted to get eyes on interface first.

### CODE REVIEW

File: src/go/trace/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,18 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
```
Comment (markjen) [inline src/go/trace/BUILD.bazel:1]: our other `BUILD.bazel` files have copyright headers

**Review by markjen — COMMENTED** [review]
should we use https://pkg.go.dev/github.com/opentracing/opentracing-go as an interface and plug sentry.io in as a `Tracer`?

File: src/go/trace/BUILD.bazel (hunk 1 above)
Comment (mstre123) [reply src/go/trace/BUILD.bazel:1]: Not sure that's true, all of the build.bazel files in the src/go repo dont have a copyright header
https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/magma/magma%24+file:src/go+file:%5C.bazel%24&patternType=literal

**Comment by ekowtaylor** [comment]
@mstre123 what's the latest with this work. Any plans to move it forward else lets close it?

**Pull request closed without merging**